code/pages/DMSDocumentCartCheckoutPage.php

```diff
@@ -0,0 +1,258 @@
+            $emailAddress = ($info = $cart->getReceiverInfo()) ? $info['ReceiverEmail'] : $from;
+            $email = Email::create(
+                $from,
+                $emailAddress,
+                _t('DMSDocumentCartCheckoutPage.EMAIL_SUBJECT', 'Request for Printed Publications')
+            );
+            $email->setBcc($member->Email);
+            $renderedCart = $cart->renderWith('DocumentCart_email');
+            $body = _t(
+                'DMSDocumentCartCheckoutPage.EMAIL_BODY',
+                '<p>A request for printed publications has been submitted with the following details:</p>'
+            );
+            $body .= $renderedCart->getValue();
+            $email->setBody($body)->setReplyTo($emailAddress);
+            $this->extend('updateSend', $email);
+            return $email->send();
+        } else {
+            throw new DMSDocumentCartException('No recipient has been configured. Please do so from the CMS');
+        }
+    }
+
+    /**
+     * Handles form submission.
+     * Totals requested are updated, delivery details added, email sent for fulfillment
+     * and print request totals updated.
+     *
+     * @param array          $data
+     * @param Form           $form
+     * @param SS_HTTPRequest $request
+     *
+     * @return SS_HTTPResponse
+     */
+    public function doRequestSend($data, Form $form, SS_HTTPRequest $request)
+    {
+        $this->updateCartItems($data);
+        $this->updateCartReceiverInfo($data);
+        $this->send();
+        $this->trackTimestampedPrintRequest();
+        $this->getCart()->saveSubmission($form);
+        $this->getCart()->emptyCart();
+        return $this->redirect("{$this->Link()}?message=1");
+    }
+
+    /**
+     * Increments the print counts of all documents which were successfully sent.
+     */
+    public function trackTimestampedPrintRequest()
+    {
+        /** @var DMSRequestItem $item */
+        foreach ($this->getCart()->getItems() as $item) {
+            /** @var DMSDocument|DMSDocumentCartExtension $doc */
+            $doc = $item->getDocument();
+            $doc->incrementPrintRequest();
+        }
+    }
+
+    /**
+     * Retrieves a {@link DMSDocumentCart} instance
+     *
+     * @return DMSDocumentCart
+     */
+    public function getCart()
+    {
+        return singleton('DMSDocumentCart');
+    }
+
+    /**
+     * Updates the document quantities just before the request is sent.
+     *
+     * @param array $data
+     */
+    public function updateCartItems($data)
+    {
+        if (isset($data['ItemQuantity']) && !empty($data['ItemQuantity'])) {
+            foreach ($data['ItemQuantity'] as $itemID => $quantity) {
+                // Only update if quantity has changed
+                $item = $this->getCart()->getItem($itemID);
+                if ($item->getQuantity() == $quantity) {
+                    continue;
+                }
+                $this->getCart()->updateItemQuantity($itemID, $quantity - 1);
```

Comment:
This is done for a different reason here. On the checkout page users can input the number of copies they desire. Adding an item to the cart adds Quantity=1. So now with the changed logic to DMSDocumentCart->updateItemQuantity() you suggested, adding a positive number adds it to the existing quantity, which means if a user changes the quantity on said checkout page from 1 to 2 it results in 3. The -1 offsets this anomaly.